internal/devices/create.go

```diff
@@ -64,62 +66,127 @@ func (c *Client) Create() *cobra.Command {
   metal device create -p $METAL_PROJECT_ID -P c3.medium.x86 -m sv -H test-rocky -O rocky_8 -r 47161704-1715-4b45-8549-fb3f4b2c32c7`,
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			var endDt *packngo.Timestamp
-
 			if userdata != "" && userdataFile != "" {
 				return fmt.Errorf("either userdata or userdata-file should be set")
 			}
 			cmd.SilenceUsage = true
-
 			if userdataFile != "" {
 				userdataRaw, readErr := os.ReadFile(userdataFile)
 				if readErr != nil {
-					return fmt.Errorf("Could not read userdata-file: %w", readErr)
+					return fmt.Errorf("could not read userdata-file: %w", readErr)
 				}
 				userdata = string(userdataRaw)
 			}
+			var endDt time.Time
 
 			if terminationTime != "" {
 				parsedTime, err := time.Parse(time.RFC3339, terminationTime)
 				if err != nil {
-					return fmt.Errorf("Could not parse time %q: %w", terminationTime, err)
+					return fmt.Errorf("could not parse time %q: %w", terminationTime, err)
 				}
-				endDt = &packngo.Timestamp{Time: parsedTime}
+				endDt = parsedTime
+			}
+			var publicSubnet int32
+			// In case of no configuration,
+			if publicIPv4SubnetSize == 0 {
+				publicSubnet = 31 // Assigin default to 31
```

Comment:
Why is a default needed for `metal-go` but not for `packngo`?

Reply:
In case of no value from the user, cmd flags read it has 0, metal-go doesn't support '0' subnet size

```
./metal device create -p dbf78a6b-b67b-4c52-9b24-740e8847cd9e -P c3.small.x86 -m da -O ubuntu_20_04 -H create-device-dev
```

Reply:
Rather than setting a default subnet here, we should trust the API to set the default.  That would be accomplished by only calling `facilityDeviceRequest.DeviceCreateInFacilityInput.SetPublicIpv4SubnetSize(int32(publicIPv4SubnetSize))` when `publicIPv4SubnetSize != 0`.